clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1034,7 +1034,7 @@ private void close(long timeout, TimeUnit timeUnit, boolean swallowException) {
                     try {
                         this.ioThread.join(timeUnit.toMillis(timeout));
                     } catch (InterruptedException t) {
-                        firstException.compareAndSet(null, t);
+                        firstException.compareAndSet(null, new InterruptException(t));
```

Comment:
Why we want to wrap the exception `t` again?

Reply:
We catch `java.lang.InterruptedException` and wrap with `org.apache.kafka.common.errors. InterruptException` as documented in JavaDocs.